platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheUnsupportedTypesIntegrationTest.groovy

```diff
@@ -311,6 +313,11 @@ class ConfigurationCacheUnsupportedTypesIntegrationTest extends AbstractConfigur
         concreteTypeName = concreteType instanceof Class ? concreteType.name : concreteType
     }
 
+    private static executorServiceTypeOnCurrentJvm() {
+        def shortName = Jvm.current().javaVersion == JavaVersion.VERSION_21 ? 'AutoShutdownDelegatedExecutorService' : 'FinalizableDelegatedExecutorService'
+        return 'java.util.concurrent.Executors$' + shortName
+    }
```

Comment:
Instead of switching out the private implementation type, could we change this test case to look at creating a `ThreadPoolExecutor` directly and show that's not supported?

It's hard to tell if we are checking that this way (through Executors) of creating a thread pool executor is unsupported or just that it's a thread pool executor. If we just care about the underlying type, we can create a thread pool executor directly.